cmd/antrea-agent/util_test.go

```diff
@@ -0,0 +1,106 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestGetAvailableNodePortAddresses(t *testing.T) {
+	testCases := []struct {
+		name                        string
+		nodePortAddressesFromConfig []string
+		excludeDevices              []string
+	}{
+		{
+			name:                        "empty nodePortAddressesFromConfig",
+			nodePortAddressesFromConfig: []string{},
+			excludeDevices:              []string{"antrea-egress0", "antrea-ingress0", "kube-ipvs0", "antrea-gw0"},
+		},
+		{
+			name:                        "good path with nodePortAddressesFromConfig",
+			nodePortAddressesFromConfig: []string{"1.2.3.0/24", "1.2.3.4/32"},
+			excludeDevices:              []string{"antrea-egress0", "antrea-ingress0", "kube-ipvs0", "antrea-gw0"},
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			_, _, err := getAvailableNodePortAddresses(tc.nodePortAddressesFromConfig, tc.excludeDevices)
+			assert.NoError(t, err)
+		})
+	}
+}
+
+func TestParsePortRange(t *testing.T) {
+	testCases := []struct {
+		name          string
+		portRangeStr  string
+		expectedstart int
+		expectedend   int
+		expectederr   string
+	}{
+		{
+			name:         "wrong port range format",
+			portRangeStr: "200 300",
+			expectederr:  "wrong port range format: 200 300",
+		},
+		{
+			name:         "length of port range string not equals to 2",
+			portRangeStr: "6110-7000-8000",
+			expectederr:  "wrong port range format: 6110-7000-8000",
+		},
+		{
+			name:         "wrong port range type value for start",
+			portRangeStr: "wrong-6200",
+			expectederr:  "strconv.Atoi: parsing \"wrong\": invalid syntax",
+		},
+		{
+			name:         "wrong port range type value for end",
+			portRangeStr: "6100-wrong",
+			expectederr:  "strconv.Atoi: parsing \"wrong\": invalid syntax",
+		},
+		{
+			name:          "valid range",
+			portRangeStr:  "61000-62000",
+			expectedstart: 61000,
+			expectedend:   62000,
+		},
+		{
+			name:         "port range's end smaller than port range's start",
+			portRangeStr: "6000-5000",
+			expectederr:  "start port must be smaller than end port: 6000-5000",
+		},
+		{
+			name:         "port range's start and port range's end with equal values",
+			portRangeStr: "6100-6100",
+			expectederr:  "start port must be smaller than end port: 6100-6100",
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			start, end, err := parsePortRange(tc.portRangeStr)
+			assert.Equal(t, tc.expectedstart, start)
+			assert.Equal(t, tc.expectedend, end)
+			if tc.expectederr != "" {
+				assert.EqualError(t, err, tc.expectederr)
+			} else {
+				assert.NoError(t, err)
+			}
```

Comment:
It makes more sense to check error first, then start and end, as checking the latters is pointless when it's expected to fail.
start and end only need to be checked when expectedErr is empty.